perception/sandbox/polygrasp/src/polygrasp/robot_interface.py

```diff
@@ -0,0 +1,151 @@
+    ):
+        states = []
+        for _ in range(max_attempts):
+            states += self.move_to_ee_pose(
+                position=position, orientation=orientation, time_to_go=time_to_go
+            )
+            curr_ee_pos, curr_ee_ori = self.get_ee_pose()
+
+            log.info(f"Dist to goal: {torch.linalg.norm(curr_ee_pos - position)}")
+
+            if (
+                states[-1].prev_command_successful
+                and torch.linalg.norm(curr_ee_pos - position) < success_dist
+            ):  # TODO: orientation diff
+                break
+        return states
+
+    def select_grasp(self, grasps: graspnetAPI.GraspGroup) -> graspnetAPI.Grasp:
+        with torch.no_grad():
+            feasible_i = []
+            for i, grasp in enumerate(grasps):
+                grasp_point, grasp_approach_delta, des_ori_quat = compute_des_pose(grasp)
+                point_a = grasp_point + self.k_approach * grasp_approach_delta
+                point_b = grasp_point + self.k_grasp * grasp_approach_delta
+
+                def check_feasibility(point):
+                    ik_sol = self.robot_model.inverse_kinematics(
+                        torch.Tensor(point), torch.Tensor(des_ori_quat)
+                    )
+                    ee_pos, ee_quat = self.robot_model.forward_kinematics(ik_sol)
+                    return torch.linalg.norm(ee_pos - point) < 0.001
+
+                if check_feasibility(point_a) and check_feasibility(point_b):
+                    feasible_i.append(i)
+
+            if len(feasible_i) < len(grasps):
+                log.warning(
+                    f"Filtered out {len(grasps) - len(feasible_i)}/{len(grasps)} grasps due to kinematic infeasibility."
+                )
+
+            # Choose the grasp closest to the neutral position
+            grasp, i = min_dist_grasp(self.default_ee_pose, grasps[feasible_i][:5])
+            log.info(f"Closest grasp to ee ori, within top 5: {i + 1}")
+            return grasp
+
+    def grasp(
+        self,
+        grasp: graspnetAPI.Grasp,
+        time_to_go=3,
+        offset=np.array([0.0, 0.0, 0.0]),
+        gripper_width_success_threshold=0.0005,
+    ):
+        states = []
+        grasp_point, grasp_approach_delta, des_ori_quat = compute_des_pose(grasp)
+
+        self.gripper_open()
+        states += self.move_until_success(
+            position=torch.Tensor(grasp_point + grasp_approach_delta * self.k_approach + offset),
+            orientation=torch.Tensor(des_ori_quat),
+            time_to_go=time_to_go,
+        )
+
+        grip_ee_pos = torch.Tensor(grasp_point + grasp_approach_delta * self.k_grasp + offset)
+
+        states += self.move_until_success(
+            position=grip_ee_pos, orientation=torch.Tensor(des_ori_quat), time_to_go=time_to_go
+        )
+        self.gripper_close()
+
+        log.info(f"Waiting for gripper to close...")
+        time.sleep(1.5)
```

Comment:
Wait until `self.gripper.get_state().is_moving == False` instead of waiting a fixed amount of time? Sleeping a fixed amount of time could cause inaccurate results if a lag spike happens or if a soft, highly-damped object takes a long time to grasp.